cpp/src/IceLocatorDiscovery/PluginI.cpp

```diff
@@ -372,56 +365,36 @@ Request::response(bool ok, const pair<const Ice::Byte*, const Ice::Byte*>& outPa
 }
 
 void
-Request::exception(const Ice::Exception& ex)
+Request::exception(std::exception_ptr ex)
 {
     try
     {
-        ex.ice_throw();
+        rethrow_exception(ex);
     }
     catch(const Ice::RequestFailedException&)
     {
-        _amdCB.second(current_exception());
+        _amdCB.second(ex);
     }
     catch(const Ice::UnknownException&)
     {
-        _amdCB.second(current_exception());
+        _amdCB.second(ex);
     }
     catch(const Ice::NoEndpointException&)
     {
-        try
-        {
-            throw Ice::ObjectNotExistException(__FILE__, __LINE__);
-        }
-        catch(...)
-        {
-            _amdCB.second(current_exception());
-        }
+
+        _amdCB.second(make_exception_ptr(Ice::ObjectNotExistException(__FILE__, __LINE__)));
     }
     catch(const Ice::CommunicatorDestroyedException&)
     {
-        try
-        {
-            throw Ice::ObjectNotExistException(__FILE__, __LINE__);
-        }
-        catch(...)
-        {
-            _amdCB.second(current_exception());
-        }
+        _amdCB.second(make_exception_ptr(Ice::ObjectNotExistException(__FILE__, __LINE__)));
     }
     catch(const Ice::ObjectAdapterDeactivatedException&)
     {
-        try
-        {
-            throw Ice::ObjectNotExistException(__FILE__, __LINE__);
-        }
-        catch(...)
-        {
-            _amdCB.second(current_exception());
-        }
+        _amdCB.second(make_exception_ptr(Ice::ObjectNotExistException(__FILE__, __LINE__)));
     }
-    catch(const Ice::Exception&)
+    catch (...)
```

Comment:
I thought we preferred catching std::exception as the most base exception? Everything should be deriving from it.

Reply:
That's correct, we assume all exceptions derive from std::exception.

However, in this particular code, we don't care: the fallback/catchall is to forward the exception. I see this as similar to:
```
catch { ... }
```
vs
```
catch (Exception) { ... }
```
in C#. 

catch { } in C# corresponds to catch (...) in C++.